### PR TITLE
Drop support for Python 3.9

### PR DESCRIPTION
**Pull request by phofl**
Closes #xxxx

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: continuous_integration/environment-3.9.yaml [hunk 1]
```diff
@@ -1,60 +0,0 @@
...
-  - pyarrow
-  - pynvml  # Only tested here
-  - pytest
-  - pytest-cov
-  - pytest-faulthandler
-  - pytest-repeat
-  - pytest-rerunfailures
-  - pytest-timeout
-  - python-snappy  # Only tested here
-  - pytorch  # Only tested here
```
Comment (fjetter) [inline continuous_integration/environment-3.9.yaml:42]: There are a couple of "only tested here" dependencies. We should add them to another environment, probably 3.10 is the best one

File: continuous_integration/recipes/distributed/meta.yaml [hunk 2]
```diff
@@ -24,28 +24,28 @@ build:
...
     - dask-core {{ dask_version }}
     - versioneer =0.29
     - tomli # [py<311]
   run:
-    - python >=3.9
+    - python >=3.10
     - click >=8.0
-    - cloudpickle >=1.5.0
-    - cytoolz >=0.10.1
+    - cloudpickle >=2.0.0
```
Comment (fjetter) [inline continuous_integration/recipes/distributed/meta.yaml:35]: This does not match with the pyproject.toml dependencies. I'm fine with bumping this (according to pypi tags, the first cloudpickle version supporting 3.10 is actually 3.0)

If this goes away, we can also drop some compat code, see https://github.com/dask/distributed/blob/879e5924ccb4f1bb45e0090f6ca9e6f7eaf343ce/distributed/protocol/pickle.py#L13

File: continuous_integration/recipes/distributed/meta.yaml [hunk 3]
```diff
@@ -24,28 +24,28 @@ build:
...
-    - psutil >=5.7.2
-    - pyyaml >=5.3.1
+    - psutil >=5.8.0
+    - pyyaml >=5.4.1
     - sortedcontainers >=2.0.5
     - tblib >=1.6.0
     - toolz >=0.10.0
-    - tornado >=6.0.4
-    - urllib3 >=1.24.3
+    - tornado >=6.2.0
```
Comment (fjetter) [inline continuous_integration/recipes/distributed/meta.yaml:47]: also doesn't match pyproject.toml

Comment (fjetter) [reply continuous_integration/recipes/distributed/meta.yaml:47]: I guess this is true for all of the above.

File: continuous_integration/recipes/distributed/meta.yaml (hunk 2 above)
Comment (jrbourbeau) [reply continuous_integration/recipes/distributed/meta.yaml:35]: > according to pypi tags, the first cloudpickle version supporting 3.10 is actually 3.0

I went ahead and bumped the dependencies in `pyproject.toml` to have a minimum of 2.0.0. Looking through the git history, I think the PyPI tags were updated after 3.10 support was actually added. Also fine with bumping to 3.0.0 if folks prefer. 

> If this goes away, we can also drop some compat code, see

Dropped 👍

File: continuous_integration/recipes/distributed/meta.yaml (hunk 3 above)
Comment (jrbourbeau) [reply continuous_integration/recipes/distributed/meta.yaml:47]: Updated

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**